addons/l10n_it_edi/models/account_move.py

```diff
@@ -1190,6 +1190,10 @@ def build_error(message, records):
             if moves := pa_moves.filtered(lambda move: move.l10n_it_origin_document_date and move.l10n_it_origin_document_date > fields.Date.today()):
                 message = _("The Origin Document Date cannot be in the future.")
                 errors['move_future_origin_document_date'] = build_error(message=message, records=moves)
+        if pa_moves := self.filtered(lambda move: len(move.commercial_partner_id.l10n_it_pa_index or '') == 7):
+            if moves := pa_moves.filtered(lambda move: not move.l10n_it_origin_document_type and move.l10n_it_cig and move.l10n_it_cup):
+                message = _("CIG/CUP fields of partner(s) are present, please fill out Origin Document Type field in the Electronic Invoicing tab.")
+                errors['move_missing_origin_document'] = build_error(message=message, records=moves)
```

Comment:
I'd use a different key here to avoid overriding

Reply:
IMO, there can't be overrides here. 
As the first setting of [`move_missing_origin_document`](https://github.com/odoo/odoo/pull/173682/files/4e477632997b34c506f9e0cc7d1b7cabe29daaa4#diff-7a314f6fb9514fd24adb9e88e00fe802ff75743c7227d648a65f44ca34de9883R1189) is in the case of Public administration(i.e. company with 6 digits).  
The second I added is in the case of a private company(i.e. with 7 digits).   
Therefore, this value can't be overridden.

Reply:
I assume self can contain many records, some of which go to a pa partner and some not?

Reply:
hmm, that makes sense 😅